modules/apps/site-initializer/site-initializer-raylife-ap/src/main/resources/site-initializer/fragments/group/src/raylife-ap-react-fragments/fragments/policies-total-premium-due/index.js

```diff
@@ -12,176 +12,341 @@
-		handleExpirationPolicies();
-	}, []);
+	const handleExpirationPolicies = () => {
+		Promise.allSettled([
+			getProducts(),
+			getPoliciesChartExpiringPolicies(
+				currentDateString[0],
+				currentDateString[1],
+				currentDateString[2],
+				lastDateCurrentMonth[0],
+				lastDateCurrentMonth[1],
+				lastDateCurrentMonth[2]
+			),
+		]).then((results) => {
+			const [
+				productPoliciesResult,
+				expiringPoliciesResultThisMonth,
+			] = results;
+			let policiesObjectList = {};
+			const productList = productPoliciesResult?.value?.data?.items;
+
+			productList?.map((product) => {
+				product.productValue = 0;
+			});
 
-	policies?.forEach((policy) => {
-		expirationPoliciesList[policy?.productName] += policy?.termPremium;
+			const productListFormatted = {};
+			productList?.forEach((product) => {
+				if (!productListFormatted[product?.name]) {
+					productListFormatted[product?.name] = 0;
+				}
 
-		if (!expirationPoliciesList[policy?.productName]) {
-			expirationPoliciesList[policy?.productName] = policy?.termPremium;
-		}
-	});
+				setExpirationPoliciesList({...productListFormatted});
+				policiesObjectList = {...productListFormatted};
+			});
+
+			if (selectedFilterDate === PERIOD.THIS_MONTH) {
+				const policiesList = [];
+
+				const totalFilteredPolicies =
+					expiringPoliciesResultThisMonth?.value?.data;
+
+				totalFilteredPolicies?.items.map((policy) => {
+					policiesList.push(policy);
+				});
+
+				const policiesObject = createPolicyObj(
+					policiesList,
+					policiesObjectList
+				);
+
+				const dataColumnsFormatted = createDataPolicyToBarChartFormat(
+					policiesObject
+				);
+				const labelColumnsFormatted = createLabelPolicyToBarChartFormat(
+					policiesObject
+				);
+
+				setDataChart(dataColumnsFormatted);
+
+				const formattedLabel = formatLabel(
+					labelColumnsFormatted,
+					maxProductNameLenght
+				);
 
-	const dataColumnsFormatted = [...Object.values(expirationPoliciesList)];
-	dataColumnsFormatted.unshift('data');
+				setLabelChart(formattedLabel);
 
-	const labelColumnsFormatted = [...Object.keys(expirationPoliciesList)];
-	labelColumnsFormatted.unshift('x');
+				createSumPolicies(dataColumnsFormatted);
 
-	const formatLabel = (labelColumnsFormatted, maxNameLenght) => {
-		const productAbbrevation = labelColumnsFormatted.map((name) => {
-			if (name?.length > maxNameLenght) {
-				return name
-					?.split(' ')
-					.map((product) => product.charAt(0))
-					.join('');
+				setIsLoading(true);
+
+				return;
 			}
 
-			return name;
-		});
+			if (selectedFilterDate === PERIOD.THREE_MONTH) {
+				getPoliciesChartExpiringPolicies(
+					currentDateString[0],
+					currentDateString[1],
+					currentDateString[2],
+					lastDateNextThreeMonth[0],
+					lastDateNextThreeMonth[1],
+					lastDateNextThreeMonth[2]
+				).then((results) => {
+					const policiesList = [];
+
+					const totalFilteredPolicies = results?.data;
+
+					totalFilteredPolicies?.items.map((policy) => {
```

Comment:
Items also need the '**?**'. Put it for everyone.

Reply:
Done.